mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -114,6 +115,32 @@ abstract class DecisionTreeTests(sc: SparkContext)
 
   protected var labelType = -1
 
+  def runTest(
+    rdd: RDD[LabeledPoint], transposedDataset: Option[RDD[(Int, Vector)]]): TreeBasedModel
+
+  override def run(): JValue = {
+    val algType: String = stringOptionValue(ALG_TYPE)
+    // Transpose dataset before timing "byCol"
+    val transposedDataset = algType match {
+      case "byRow" => None
+      case "byCol" => Some(TreeUtil.rowToColumnStoreDense(rdd.map(_.features)))
```

Comment:
You should persist this and then count it here.  That way, it will be materialized before running the test.

Also, rather than calling cache(), I'd persist() to MEMORY_AND_DISK to make it a little more robust.